common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationSettings.java

```diff
@@ -255,6 +255,15 @@ public void setDeviceCertificateProxyClass(@SuppressWarnings(WarningType.rawtype
         return mClazzDeviceCertProxy;
     }
 
+
+    /**
+     * remove class for work place join related API. This is only used from
+     * Authenticator side.
+     */
+    public void removeDeviceCertificateProxy() {
```

Comment:
This is definitely strange... likely some additional comments here would help reviewer understand why this is required.